File: c/fizzbuzz.c
#include <stdio.h>

int
main(int argc, char **argv) {
	int max = 100;
	int i;

	if (argc > 1)
		max = atoi(argv[1]);
	for (i = 1; i <= max; ++i) {
		if (i % 3 == 0) {
			printf("Fizz");
		}
		if (i % 5 == 0) {
			printf("Buzz");
		}
		if ((i % 3) && (i % 5)) {
			printf("%d", i);
		}
		printf("\n");
	}
}

File: rust/fizzbuzz.rs
use std::env;

fn main() {
    let args: Vec<String> = env::args().collect();
    let mut max = 101;

    if args.len() > 1 {
        let arg = args[1].parse::<i32>();
	if arg.is_ok() {
	    max = arg.unwrap() + 1;
	}
    }

    for n in 1..max {
	if n % 15 == 0 {
	    println!("FizzBuzz");
	} else if n % 3 == 0 {
	    println!("Fizz");
	} else if n % 5 == 0 {
	    println!("Buzz");
	} else {
	    println!("{}", n);
	}
    }
}
